tests/Agent/IntegrationTests/IntegrationTestHelpers/RandomPortGenerator.cs

```diff
@@ -32,21 +32,27 @@ public static int NextPort()
         {
             lock (_usedPortLock)
             {
-                for (var countAttempts = 0; countAttempts < maxAttempts; countAttempts++)
+                var countAttempts = 0;
```

Comment:
Since the ports don't actually need to be random, they just need to be unique, I think we could come up with a way to block off ports that would minimize conflicts. Something based on the timestamp and an incremental counter, for example. That can be a future optimization, though. I don't have it fully fleshed out in my head.

Reply:
I definitely took the easy route here and just bolted on ignoring the previous ports.